.github/CONTRIBUTING.md

```diff
@@ -0,0 +1,117 @@
+# Contributing Guidelines
+
+## Quick Start
+
+### Ideas
+
+- File an [issue][issues].
+- Explain why you want the feature. How does it help you? What for do you want the feature?
+
+### Bugs
+
+- File an [issue][issues].
+- Ideally, write a failing test and send it as a Pull Request.
+
+### Coding
+
+- Marvin is written in [Python][].
+- Marvin uses [Semantic Release](#sem-rel).
+
+#### Recommended Workflow
+
+1. Fork Marvin.
+2. Create a feature branch.
+3. Execute `source bin/setup-env`
+4. Write tests.
+5. Write code.
+6. test what you created: `pytest tests/`
+7. Send a Pull Request.
+8. Make sure [test coverage][] didn't drop and all CI builds are passing.
+
+<a name="sem-rel"></a>
+#### Semantic Release and Conventional Changelog
+
+Releasing of new Marvin versions to [Pypi][] is automatically managed by [Semantic Release][].
+Semantic Release makes sure correct version numbers get bumped according to the **meaning**
+of your changes once your PR gets merged to `master`.
+
+To make it work, it's necessary to follow [Conventional Commits][]. That basically
+means all commit messages in the project should follow a particular format:
+
+```
+<type>: <subject>
+```
+
+Where `<type>` is:
+
+- `feat` - New functionality added
+- `fix` - Broken functionality fixed
+- `perf` - Performance improved
+- `docs` - Documentation added/removed/improved/...
+- `chore` - Package setup, CI setup, ...
+- `refactor` - Changes in code, but no changes in behavior
+- `test` - Tests added/removed/improved/...
+
+In the rare cases when your changes break backwards compatibility, the message
+must include string `BREAKING CHANGE:`. That will result in bumping the major version.
+
+Seems hard?
+
+- See [existing commits][] as a reference
+- A `commit-msg` git hook validates the format of your messages
+
+### Testing
+
+Use `pytest tests/` to run all tests on your default python platform.
+Use `tox` to run the tests in all supported platforms.
+
+### Linting
+
+Marvin uses [flake8][] to lint both Marvin and tests codebases. Execute `flake8 marvin tests`.
+
+### Documentation
+
+The main documentation is written in [Markdown][]. Marvin uses
+[ReadTheDocs][] to build and publish the documentation:
+
+- [https://marvin.readthedocs.io](https://marvin.readthedocs.io) - preferred long URL
+- [http://marvin.rtfd.org](http://marvin.rtfd.org) - preferred short URL
```

Comment:
I don't know, I just wrote those as placeholders. We should register an account there.